### PR TITLE
Add setLocalMobileUxConfiguration.

### PR DESCRIPTION
**Pull request by ianlin-bbpos**
## Summary

Add new feature of LocalMobileUxConfiguration.

## Motivation

To support setting LocalMobileUxConfiguration

## Testing

- [x] I tested this manually
- [ ] I added automated tests

## Documentation

Select one:

- [ ] I have added relevant documentation for my changes.
- [x] This PR does not result in any developer-facing changes.

### CODE REVIEW

File: android/src/main/java/com/stripeterminalreactnative/StripeTerminalReactNativeModule.kt [hunk 1]
```diff
@@ -1082,6 +1082,51 @@ class StripeTerminalReactNativeModule(reactContext: ReactApplicationContext) :
...
+        var tapZone: LocalMobileUxConfiguration.TapZone? = null
+        val tapZoneParam = params.getMap("tapZone")
+        tapZoneParam?.let {
+            val tapZoneIndicator = mapToTapZoneIndicator(tapZoneParam.getString("tapZoneIndicator"))
+
+            val tapZonePosition = tapZoneParam.getMap("tapZonePosition")?.let {
+                LocalMobileUxConfiguration.TapZonePosition.Manual(
+                    it.getDouble("xBias").toFloat(),
+                    it.getDouble("yBias").toFloat())
+            }
```
Comment (chr-stripe) [inline android/src/main/java/com/stripeterminalreactnative/StripeTerminalReactNativeModule.kt:1099]:
```suggestion
            } ?: TapZonePosition.Default
```

File: android/src/main/java/com/stripeterminalreactnative/StripeTerminalReactNativeModule.kt [hunk 2]
```diff
@@ -1082,6 +1082,51 @@ class StripeTerminalReactNativeModule(reactContext: ReactApplicationContext) :
...
+
+            val tapZoneBuilder = LocalMobileUxConfiguration.TapZone.Manual.Builder()
+            tapZoneIndicator.let { tapZoneBuilder.indicator(it) }
+            tapZonePosition?.let { tapZoneBuilder.position(it) }
+            tapZone = tapZoneBuilder.build()
```
Comment (chr-stripe) [inline android/src/main/java/com/stripeterminalreactnative/StripeTerminalReactNativeModule.kt:1104]:
```suggestion
            tapZone = LocalMobileUxConfiguration.TapZone.Manual.Builder()
                .indicator(tapZoneIndicator)
                .position(tapZonePosition)
                .build()
```

File: android/src/main/java/com/stripeterminalreactnative/Mappers.kt [hunk 3]
```diff
@@ -888,3 +889,35 @@ fun mapFromCollectDataType(type: String): CollectDataType? {
...
+        else -> LocalMobileUxConfiguration.TapZoneIndicator.DEFAULT
+    }
+}
+
+fun mapToDarkMode(mode: String?): LocalMobileUxConfiguration.DarkMode {
+    return when (mode) {
+        "dark" -> LocalMobileUxConfiguration.DarkMode.DARK
+        "light" -> LocalMobileUxConfiguration.DarkMode.LIGHT
+        "system" -> LocalMobileUxConfiguration.DarkMode.SYSTEM
+        else -> LocalMobileUxConfiguration.DarkMode.SYSTEM
```
Comment (chr-stripe) [inline android/src/main/java/com/stripeterminalreactnative/Mappers.kt:909]: I believe the default is actually:

```suggestion
        else -> LocalMobileUxConfiguration.DarkMode.LIGHT
```

File: android/src/main/java/com/stripeterminalreactnative/StripeTerminalReactNativeModule.kt [hunk 4]
```diff
@@ -1082,6 +1082,51 @@ class StripeTerminalReactNativeModule(reactContext: ReactApplicationContext) :
...
+        }
+        localMobileUxConfigurationBuilder.tapZone(tapZone?:LocalMobileUxConfiguration.TapZone.Default)
+
+        val colorsParam = params.getMap("colors")
+        colorsParam?.let {
+            val colorSchemeBuilder = LocalMobileUxConfiguration.ColorScheme.Builder()
+            colorSchemeBuilder.apply {
+                primary(it.getString("primary")?.let {
+                    LocalMobileUxConfiguration.Color.Value(hexToArgb(it)) } ?:
+                    LocalMobileUxConfiguration.Color.Default)
```
Comment (chr-stripe) [inline android/src/main/java/com/stripeterminalreactnative/StripeTerminalReactNativeModule.kt:1114]: nit: you can probably pull this out into a private extension method for re-use

```kotlin
private fun String?.toLocalMobileColor(): LocalMobileUxConfiguration.Color {
    return this
        ?.let { LocalMobileUxConfiguration.Color.Value(hexToArgb(it)) }
        ?: LocalMobileUxConfiguration.Color.Default
}
```

And then call it like:

```kotlin
colorSchemeBuilder.apply {
    primary(it.getString("primary").toLocalMobileColor())
    success(it.getString("success").toLocalMobileColor())
    error(it.getString("error").toLocalMobileColor())
}
```

File: src/types/index.ts [hunk 5]
```diff
@@ -517,3 +517,39 @@ export type CollectDataResultType =
       collectedData?: undefined;
       error: StripeError;
     };
+
+export type LocalMobileUxConfiguration = {
```
Comment (chr-stripe) [inline src/types/index.ts:521]: I think the properties in `LocalMobileUxConfiguration`, `TapZone`, and `Colors` should probably all be nullable? Or some other way to allow users to use defaults. The mapper methods all look like they're set up to handle null values but I think this definition requires them to be specified?

File: android/src/main/java/com/stripeterminalreactnative/StripeTerminalReactNativeModule.kt (hunk 1 above)
Comment (ianlin-bbpos) [reply android/src/main/java/com/stripeterminalreactnative/StripeTerminalReactNativeModule.kt:1099]: Oops, I missed TapZonePosition.Default ..

File: src/types/index.ts (hunk 5 above)
Comment (ianlin-bbpos) [reply src/types/index.ts:521]: yes, that make sense.

File: src/types/index.ts [hunk 6]
```diff
@@ -517,3 +517,39 @@ export type CollectDataResultType =
...
+  BELOW = 'below',
+  FRONT = 'front',
+  BEHIND = 'behind',
+}
+
+export type Colors = {
+  primary: string;
+  success: string;
+  error: string;
+};
```
[the remark this answers is not in the document]
Comment (ianlin-bbpos) [reply src/types/index.ts:549]: I've tried use Int, but got a diff value on native side after transfer, seems like a overflow.

**Review by chr-stripe — APPROVED** [review]

**Review by nazli-stripe — APPROVED** [review]

**Pull request merged**